Meter.c

```diff
@@ -288,12 +289,16 @@ static const char* const GraphMeterMode_dotsAscii[] = {
 
 static void GraphMeterMode_draw(Meter* this, int x, int y, int w) {
    const ProcessList* pl = this->pl;
-
-   if (!this->drawData) {
-      this->drawData = xCalloc(1, sizeof(GraphData));
+   GraphData* data = &this->drawData;
+
+   if ((unsigned int)w * 2 > data->nValues) {
+      unsigned int oldNValues = data->nValues;
+      data->nValues = MAXIMUM(oldNValues + (oldNValues / 2), (unsigned int)w * 2);
+      data->values = xReallocArray(data->values, data->nValues, sizeof(*data->values));
+      memmove(data->values + (data->nValues - oldNValues), data->values, oldNValues * sizeof(*data->values));
+      memset(data->values, 0, (data->nValues - oldNValues) * sizeof(*data->values));
    }
```

Comment:
Strictly speaking this should at least place an `assert(w > 0);` before casting.

Prefer `size_t` over `unsigned int`.

Reply:
> Strictly speaking this should at least place an assert(w > 0); before casting.

Okay, sure.

> Prefer `size_t` over `unsigned int`.

See above.